ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +562,233 @@ const triggerSet: TriggerSet<Data> = {
+            dir3 = 'front';
+          }
+          // if the tempest was only in or out, but the sand pattern needs right or left
+          if (!dir2 && !(safeSpot & 0b010_010_010_010)) {
+            // we also need to check the other selected direction (front/back)
+            if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+              dir2 = 'right';
+            } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+              dir2 = 'left';
+            }
+          }
+          // swap in case dir2 is null, and its more natural phrasing
+          [dir2, dir3] = [dir3, dir2];
+          data.ttokSandOrbOnSet++;
+        }
+        if (dir2 && dir3) {
+          return output.triple!({
+            inOut: output[inOut]!(),
+            dir2: output[dir2]!(), // front or back
+            dir3: output[dir3]!(), // right or left
+          });
+        }
+        if (dir2) {
+          return output.double!({ inOut: output[inOut]!(), dir2: output[dir2]!() });
+        }
+        return output[inOut]!();
+      },
+      outputStrings: ttokrroneTempestSandspoutOutputStrings,
+    },
+    // Aoe
+    {
+      id: 'Hunt Ttokrrone Touchdown',
+      type: 'StartsUsing',
+      netRegex: { id: '91DB', source: 'Ttokrrone', capture: false },
+      durationSeconds: 4,
+      response: Responses.aoe(),
+      run: (data) => {
+        // Reset everything as orb sequences start with a touchdown.
+        data.ttokSandOrbSets = 0;
+        data.ttokSandOrbOnSet = 0;
+        data.ttokSandOrbs = [];
+        data.ttokSandOrbPatterns = [];
+        data.ttokRotated = 0;
+      },
+    },
+    // Front cleave, rotates C or CCW and cleaves 4 or 7 times (maybe health dependent?)
+    {
+      id: 'Hunt Ttokrrone Fangward Dustdevil',
+      type: 'StartsUsing',
+      netRegex: { id: ['91C9', '91C5'], source: 'Ttokrrone', capture: false },
+      durationSeconds: 6,
+      suppressSeconds: 1,
+      alertText: (_data, _matches, output) => output.outOfHitbox!(),
+      infoText: (_data, _matches, output) => {
+        return output.rotateFront!();
+      },
+      outputStrings: ttokrroneDustdevilOutputStrings,
+    },
+    // Rear cleave (boss turns to do front cleaves) then rotates C or CCW and cleaves 4 or 7 times (?)
+    {
+      id: 'Hunt Ttokrrone Tailward Dustdevil',
+      type: 'StartsUsing',
+      netRegex: { id: ['91CA', '91C6'], source: 'Ttokrrone', capture: false },
+      durationSeconds: 6,
+      suppressSeconds: 1,
+      alertText: (_data, _matches, output) => output.outOfHitbox!(),
+      infoText: (_data, _matches, output) => {
+        return output.rotateRear!();
+      },
+      outputStrings: ttokrroneDustdevilOutputStrings,
+    },
+    // Dashes around 6 times
+    {
+      id: 'Hunt Ttokrrone Landswallow',
+      type: 'StartsUsing',
+      netRegex: { id: '96F2', source: 'Ttokrrone', capture: false },
+      durationSeconds: 15,
+      infoText: (_data, _matches, output) => output.dodgeToRight!(),
```

Comment:
```suggestion
      infoText: (_data, _matches, output) => output.dodge!(),
```

Admittedly this is very nitpicky, but not sure we need 'toRight' as part of the output property name -- since the safe side can be either left or right.